lib/discourse_perspective.rb

```diff
@@ -163,7 +163,7 @@ def self.should_check_post?(post)
     # default not to check secured categories
     return false if !SiteSetting.perspective_check_secured_categories && post&.topic&.category&.read_restricted?
     # don't check trashed topics
-    return false if post&.topic.trashed?
+    return false if post&.topic&.trashed?
```

Comment:
No need to continue if the topic is gone

```suggestion
    return false if !post.topic || post.topic.trashed?
```